### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Dashaverz**
[DEMO LINK](https://Dashaverz.github.io/react_todo-app-with-api/)

### CODE REVIEW

**Review by GarryRocksman — APPROVED** [review]
GJ 🔥  just revise how your notifications work. Its should dessapear after 3 seconds, but its not

and one more !!!

> Add the ability to toggle the completed status of all the todos with the toggleAll checkbox:

> toggleAll button should have active class only if all the todos are completed;
> toggleAll click changes its status to the oppsite one, and set it to all the todos;
> it should work the same as several individual updates of the todos which statuses were actually changed;
> do send requests for the todos that were not changed;

**this feature doesn't work**

**Comment by GarryRocksman** [comment]
and one more 
 - toggleAll button should have active class only if all the todos are completed; !!! This is important.

File: src/App.tsx [hunk 1]
```diff
@@ -1,24 +1,195 @@
...
+    = useState<FilterStatus>(FilterStatus.All);
+  const [tempTodo, setTempTodo] = useState('');
+  const [isClearCompleted, setIsClearCompleted] = useState(false);
+  const [errorType, setErrorType] = useState<ErrorType>(ErrorType.None);
+
+  const [isToggled, setIsToggled] = useState(false);
+
+  const userId = useContext(UserContext);
+
+  const getTodosFromServer = useCallback(async () => {
```
Comment (ArturErmolenko) [inline src/App.tsx:36]: lets handle loading here

File: src/App.tsx [hunk 2]
```diff
@@ -1,24 +1,195 @@
...
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
```
Comment (ArturErmolenko) [inline src/App.tsx:154]: add Loader while waiting todos from server

File: src/components/Footer/Footer.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+const filterOptions = Object.values(FilterStatus);
+
+export const Footer: React.FC<Props> = React.memo(({
+  activeTodosAmount,
+  filterByStatus,
+  isDone,
+  setFilterByStatus,
+  onClearCompleted,
+}) => {
+  return (
```
Comment (ArturErmolenko) [inline src/components/Footer/Footer.tsx:22]: redundant return

**Review by ArturErmolenko — CHANGES_REQUESTED** [review]
Pay attention on comment ☝️

**Pull request closed without merging**